### PR TITLE
Fix: Sidebar only renders initially after file info is returned

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

**Comment by jeremypress** [comment]
This PR addresses the issue where we initially render the loading sidebar, the files call comes back, and since there are no tabs to show the sidebar disappears.

Brad's desired behavior:
1. Initially don't show anything until we know we have at least 1 tab to show
2. When navigating between files, it's ok to show the loading sidebar, and then hide it if there's nothing to show.

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #583

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ContentSidebar.js [hunk 1]
```diff
@@ -98,6 +99,7 @@ class ContentSidebar extends PureComponent<Props, State> {
 
     initialState: State = {
         isVisible: true,
+        isNavigating: false,
```
[the remark this answers is not in the document]
Comment (jeremypress) [reply src/components/ContentSidebar/ContentSidebar.js]: This was requested by Brad. We may want to make this a prop, but we won't initially show any loading state until the files call comes back with this change.

File: src/components/ContentSidebar/ContentSidebar.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ type Props = {
 type State = {
     view?: SidebarView,
     file?: BoxItem,
-    isVisible: boolean,
+    isVisible?: boolean,
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ContentSidebar.js]: You can make this not be optional, and set it in the constructor

Comment (jeremypress) [reply src/components/ContentSidebar/ContentSidebar.js]: Flow says I can't :(

**Comment by jeremypress** [comment]
@DanDeMicco pointed out that isVisible was doing almost exactly what we needed. By flipping it to false by default we won't render anything until the files call comes back. @priyajeet are there any edge cases you can think of?

File: src/components/ContentSidebar/__tests__/ContentSidebar-test.js [hunk 3]
```diff
@@ -423,7 +423,6 @@ describe('components/ContentSidebar/ContentSidebar', () => {
         let wrapper;
         let instance;
         let fetchFileSuccessCallback;
-        let fetchFileErrorCallback;
```
Comment (jeremypress) [inline src/components/ContentSidebar/__tests__/ContentSidebar-test.js]: Cleanup from detailsSidebar refactor.
